### PR TITLE
Rename copying to license and upgrade the setup.cfg

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Description

We did the same in astroid, so here goes consistency.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -7,15 +7,15 @@ long_description_content_type = text/x-rst
 url = https://github.com/PyCQA/pylint
 author = Python Code Quality Authority
 author_email = code-quality@python.org
-license = GPL-2.0-or-later
+license = GPL-2.0
```
Comment (cdce8p) [inline setup.cfg:10]: I believe `GPL-2.0-or-later` was correct . The license explicitly states
```
... either version 2 of the License, or (at your option) any later version. ...
```

File: ChangeLog [hunk 2]
```diff
@@ -25,6 +25,8 @@ Release date: Undefined
   only contain ``__version__`` (also accessible with ``pylint.__version__``), other meta-information are still
   accessible with ``import importlib;metadata.metadata('pylint')``.
 
+* COPYING has been renamed to LICENSE for standardization.
```
Comment (cdce8p) [inline ChangeLog:28]: It might be a good idea to mention this in the `What's New` section as well.

**Review by cdce8p — COMMENTED** [review]
Sorry for being late, just saw it

File: setup.cfg (hunk 1 above)
Comment (Pierre-Sassoulas) [reply setup.cfg:10]: https://github.com/PyCQA/pylint/commit/9d959abd39cd787fa7b5d68327d19d3972fe888e

**Pull request merged**